vagrant/provisioning/roles/mongodb/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+- name: configure Mongodb 4.0 repository
+  become: yes
+  copy:
+    dest: /etc/yum.repos.d/mongodb-org.repo
+    force: no
+    content: |
+      [mongodb-org-4.0]
+      name=MongoDB Repository
+      baseurl=https://repo.mongodb.org/yum/redhat/$releasever/mongodb-org/4.0/x86_64/
+      gpgcheck=1
+      enabled=1
+      gpgkey=https://www.mongodb.org/static/pgp/server-4.0.asc
+
+- name: install Mongodb
+  become: yes
+  yum:
+    state: installed
+    disable_gpg_check: yes
+    name:
+      - mongodb-org
+  register: mongodb_install
+
+- name: ensure Mongodb is started
+  become: yes
+  systemd:
+    name: mongod
+    state: started
+
+- name: ensure Mongodb is stopped
+  become: yes
+  systemd:
+    name: mongod
+    state: stopped
+
+- name: Mongodb data and log folder structure
+  become: yes
+  file:
+    path: "{{ item.name }}"
+    state: directory
+    group: mongod
+    owner: mongod
+    setype: "{{ item.setype }}"
+  loop:
+    - name: "{{ mongodb_data_folder }}"
```

Comment:
If we follow our approach for creating folder structure for other services, here under the loop instead of defining new fact in the facts file you should fill an absolute path. For ex. - name: "{{ root_folder }}/data/mongo"